extension/src/popup.jsx

```diff
@@ -114,6 +120,37 @@ const terms = (state = false, action) => {
   }
 };
 
+const thanks_messages = [
+  "thanks1", // "Thanks! 🙂",
+  "thanks2", // "Keep it up!",
+  "thanks3", // "We couldn't do it without you!",
+  "thanks4", // "Thanks for helping out!",
+  "thanks5" //  "Way to go! You've categorized $count$ ads so far."
+];
+const thanks = (state = null, action) => {
+  switch (action.type) {
+    case SAY_THANKS:
+      // we thank you frequently at first and exponentially less often later
+      // for instance, your first ad, we have a 50% chance of thanking you
+      // and your 4th ad a 25% chance and so on.
+      let message =
+        Math.random() < 1 / (2 * Math.sqrt(action.ratings_length || 1))
```

Comment:
so much fun!

Reply:
I wish I still had my TI-82 to try out the various decay functions; I guess you can never go home again.